modules/users/tests/server/user.server.routes.tests.js

```diff
@@ -262,6 +262,601 @@ describe('User CRUD tests', function () {
     });
   });
 
+  it('forgot password should return 400 for non-existent username', function (done) {
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: 'some_username_that_doesnt_exist'
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('No account with that username has been found');
+          return done();
+        });
+    });
+  });
+
+  it('forgot password should return 400 for no username provided', function (done) {
+    var provider = 'facebook';
+    user.provider = provider;
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: ''
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('Username field must not be blank');
+        });
+
+        agent.post('/api/auth/forgot')
+          .expect(400)
+          .end(function (err, res) {
+            // Handle error
+            if (err) {
+              return done(err);
+            }
+
+            res.body.message.should.equal('Username field must not be blank');
+            return done();
+          });
+
+    });
+  });
+
+  it('forgot password should return 400 for non-local provider set for the user object', function (done) {
+    var provider = 'facebook';
+    user.provider = provider;
+    user.roles = ['user'];
+
+    user.save(function (err) {
+      should.not.exist(err);
+      agent.post('/api/auth/forgot')
+        .send({
+          username: user.username
+        })
+        .expect(400)
+        .end(function (err, res) {
+          // Handle error
+          if (err) {
+            return done(err);
+          }
+
+          res.body.message.should.equal('It seems like you signed up using your ' + user.provider + ' account');
+          return done();
+        });
+    });
+  });
+
+  it('forgot password should return 400 for user password reset request', function (done) {
+    // General comment about this test-
```

Comment:
What's Mandrill? How is that going to fit in with end user cloning the project and running our tests?